### ISSUE TITLE
Destroy - azurerm_container_group is failing

### ISSUE THREAD

**Issue by kgopi1** [hashicorp/terraform-provider-azurerm#16824]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.13.7

### AzureRM Provider Version

3.0.0

### Affected Resource(s)/Data Source(s)

azurerm_container_group

### Terraform Configuration Files

```hcl
## due to image credential doesnt support -managed id , https://github.com/hashicorp/terraform-provider-azurerm/issues/15915

resource "azurerm_container_group" "bindaci" {
  for_each = toset(var.bind_hosts)
  location            = azurerm_resource_group.bind_aci.location
  name                = "${each.value}-${var.location}-${var.environment}-aci"
  os_type             = "Linux"
  resource_group_name = azurerm_resource_group.bind_aci.name
  network_profile_id  = azurerm_network_profile.aci_network_profile[each.value].id
  ip_address_type     = "Private"
  tags                = local.default_tags
  image_registry_credential {
    password = var.password
    server   = var.server
    username = var.username
  }
  depends_on = [
    azurerm_role_assignment.aci_role, azurerm_storage_share.fileshare
  ]
  identity {
    type         = "UserAssigned"
    identity_ids = [azurerm_user_assigned_identity.aci_msi.id]
  }
  dynamic "container" {
    for_each = var.containers
    content {
      cpu    = container.value.cpu
      image  = "mylab.azurecr.io/${container.value.image}"
      memory = container.value.memory
      name   = "${container.key}-container-${var.environment}"
      dynamic "ports" {
        for_each = var.container_ports
        content {
          port     = ports.value.port
          protocol = ports.value.protocol
        }
      }
      dynamic "volume" {
        for_each = var.container_volumes
        content {
          mount_path           = volume.value.mount_path
          name                 = "${each.value}-${volume.value.name}"
          storage_account_name = data.azurerm_storage_account.stgaci.name
          storage_account_key  = data.azurerm_storage_account.stgaci.primary_access_key
          share_name           = "${each.value}-${volume.value.name}"        
          }
      }
    

      volume {
        mount_path = "/etc/bind"
        name       = "namedconf"
        secret = {
          "named.conf" = base64encode(data.template_file.bindinit[each.value].rendered)
        }
      }
    }
  }
  dynamic "exposed_port" {
    for_each = var.container_ports
    content {
      port     = exposed_port.value.port
      protocol = exposed_port.value.protocol
    }
  }
}
```

### Debug Output/Panic Output

```shell
waiting for container Group "aci" (Resource Group "RG") to finish deleting: couldnt find resource(21 retries)
```

### Expected Behaviour

Destroyed complete.

### Actual Behaviour

Error waiting for container Group "aci" (Resource Group "RG") to finish deleting: couldnt find resource(21 retries)

### Steps to Reproduce

terraform apply to create aci . 

terraform destroy to delete aci 

### Important Factoids

nil

### References

nil

**Comment by lessthanseventy** [comment]
:+1:

**Comment by lonegunmanb** [comment]
Hi @kgopi1 , I believe this issue has been fixed by #17115 . Would you please test it with v3.10.0 again? Thanks.

**Comment by kgopi1** [comment]
Yes , i tested and it destroys now .. thanks .
